src/Symfony/Component/Serializer/Tests/Normalizer/AbstractObjectNormalizerTest.php

```diff
@@ -1025,6 +1025,23 @@ protected function createChildContext(array $parentContext, string $attribute, ?
 
         $this->assertFalse($normalizer->childContextCacheKey);
     }
+
+    public function testNormalizationWithMaxDepthOnStdclassObjectDoesNotThrowWarning()
+    {
+        $object = new \stdClass();
+        $object->string = 'yes';
+
+        $classMetadataFactory = new ClassMetadataFactory(new AttributeLoader());
+        $normalizer = new ObjectNormalizer($classMetadataFactory);
+        $normalized = $normalizer->normalize($object, 'json', [
+            'json_encode_options' => 15, // JsonResponse::DEFAULT_ENCODING_OPTIONS
+            AbstractNormalizer::CIRCULAR_REFERENCE_HANDLER => static fn () => null,
+            AbstractObjectNormalizer::PRESERVE_EMPTY_OBJECTS => true,
```

Comment:
These options are unrelated to the issue and could be removed for the sake of simplicity, right?

Reply:
I'll test removing them and see if the warning is still thrown.

Reply:
I was able to remove those, thanks!